src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx

```diff
@@ -3,24 +3,39 @@ import React, {forwardRef, useCallback} from 'react';
 import {View} from 'react-native';
 // eslint-disable-next-line no-restricted-imports
 import type {ScrollView as ScrollViewRN} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
+import * as Expensicons from '@components/Icon/Expensicons';
+import MenuItem from '@components/MenuItem';
 import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
 import ScrollView from '@components/ScrollView';
 import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import Navigation from '@libs/Navigation/Navigation';
+import * as PolicyUtils from '@libs/PolicyUtils';
 import CONST from '@src/CONST';
 import ROUTES from '@src/ROUTES';
-import type {ApprovalWorkflowOnyx} from '@src/types/onyx';
+import type {ApprovalWorkflowOnyx, Policy} from '@src/types/onyx';
 import type {Approver} from '@src/types/onyx/ApprovalWorkflow';
 
 type ApprovalWorkflowEditorProps = {
+    /** The approval workflow to display */
     approvalWorkflow: ApprovalWorkflowOnyx;
+
+    /** Function to remove the approval workflow */
+    removeApprovalWorkflow?: () => void;
+
+    /** The policy for the current route */
+    policy: OnyxEntry<Policy>;
+
+    /** The policy ID */
     policyID: string;
 };
 
-function ApprovalWorkflowEditor({approvalWorkflow, policyID}: ApprovalWorkflowEditorProps, ref: ForwardedRef<ScrollViewRN>) {
+function ApprovalWorkflowEditor({approvalWorkflow, removeApprovalWorkflow, policy, policyID}: ApprovalWorkflowEditorProps, ref: ForwardedRef<ScrollViewRN>) {
```

Comment:
Let's remove policyID props and use policy.ID

Reply:
policy.policyID can be undefined which is making this so much harder, will leave it as is for now.